src/classes/UTIL_OrgTelemetry_SVC.cls

```diff
@@ -33,21 +33,219 @@
+* Specific PII info is never queried or returned by this class.
 */
 public without sharing class UTIL_OrgTelemetry_SVC {
 
     /**
-    * @description Submit specific feature enablement statistics back to the License Management Org to assist
-    * with understanding how our customers are using our functionality. There's no way to unit test this though.
-    * Each feature below must be matched to a SubscriberToLmo feature of the same name.
-    * This should called in two places: 1) When the Settings page is loaded; 2) When a tracked area is saved.
-    * FeatureManagement operations are subject to MIXED_DML errors and must be conducted asynchronously
-    */
-    @future
-    public static void submitFeatureTelemetryToLMO() {
+     * @description A complete list of the Telemetry Types (at a high level) that are handled by this service class.
+     * These are used by the batch job to process feature parameters in asynchronous groups to avoid governor
+     * limit issues. The concept is that by separating each of these into smaller groupings that are processed
+     * as separate child batch iterations, the entire process can complete even if for some reason one of the queries
+     * against a table (such as a really huge Opportunity object) were to fail.
+     */
+    public Enum TelemetryType {
+        NPSP_Settings,
+        Org_Environment,
+        Data_MaxNumRelatedOpps,
+        Data_Count_of_RD_Opps_All,
+        Data_Count_of_RD_Opps_OpenEnded,
+        Data_CountOfErrorLog
+    }
+
+    /**
+     * @description Execute the batch job that will process each telemetry featureParameter type one at a time
+     * as individual batch iterations. This process has to be asynchronous to handle governor limits against
+     * huge queries as well as to avoid MixedDML errors in case a calling process were to modify a record.
+     */
+    public static void dispatchTelemetryBatchJob() {
+        Database.executeBatch(new UTIL_OrgTelemetry_BATCH(), 1);
+    }
+
+    /**
+     * @description Called by the NPSP Settings Page to subject the current state of NPSP Settings related telemetry
+     * data if it has been changed in the NPSP Settings page.
+     * @param telemetryType
+     */
+    @Future
+    public static void processNPSPSettingTelemetry_ASync() {
+        UTIL_OrgTelemetry_SVC telemetrySvc = new UTIL_OrgTelemetry_SVC();
+        telemetrySvc.processTelemetryType(TelemetryType.NPSP_Settings);
+    }
+
+    /**
+     * @description Called by the batch job to process an individual telemetry type
+     * Submits specific feature teletry statistics back to the License Management Org to assist
+     * with understanding how our customers are using NPSP functionality.
+     * Each feature below must be matched to a SubscriberToLmo feature of the same name.
+     * @param telemetryType
+     */
+    public void processTelemetryType(TelemetryType telemetryType) {
+
+        switch on (telemetryType) {
```

Comment:
Great use of the `switch`, the method is readable after using separate methods for each case.